lib/TPP/Conversion/ConvertLinalgToXsmm/ConvertLinalgToXsmm.cpp

```diff
@@ -1069,8 +1075,13 @@ struct ConvertGenericToVnniBrgemm : public OpRewritePattern<linalg::GenericOp> {
 
     int64_t m = bufferC.getType().cast<ShapedType>().getShape()[0];
     int64_t n = bufferC.getType().cast<ShapedType>().getShape()[1];
-    int64_t k = bufferA.getType().cast<ShapedType>().getShape()[2];
-    int64_t batch = bufferA.getType().cast<ShapedType>().getShape()[0];
+    int64_t kPos = 1;
+    if (hasBatch)
+      kPos++;
+    int64_t k = bufferA.getType().cast<ShapedType>().getShape()[kPos];
+    int64_t batch = std::numeric_limits<int64_t>::max();
```

Comment:
I don't understand why `batch = MAX_INT` is "no batch". The code that uses this makes a check and doesn't use it so it could very well be just zero and we wouldn't even need the extra flag.

Reply:
Indeed we set "batch" to MAX_INT if the detected operation does not have the batch dimension (hasBatch = false). We pass the batch dimension to https://github.com/plaidml/tpp-mlir/blob/99af3d548226ad5ff0082fd6ff8359ef1b8494c5/lib/TPP/Conversion/ConvertLinalgToXsmm/ConvertLinalgToXsmm.cpp#L443. Do you see a way to avoid the flag or the batch value?

Reply:
We could also set batch to zero and not max_int.

Reply:
Yup, my point above. :)

Reply:
Maybe you cannot see from the diff but `batch` is actually used and passed to `replaceOpWithGemmLikeOp` later on.

Reply:
I did see it and it's precisely `replaceOpWithGemmLikeOp` that can change, since `batch` inside it is only used if `hasBatch` is true, and that is set to true in the line you mentioned. This can all be avoided if `batch` is zero and then you don't need the flag inside `replaceOpWithGemmLikeOp`, just do `if (batch) { ... }`.

Reply:
https://github.com/plaidml/tpp-mlir/pull/837/commits/27299c73e7e9ea7476c5a339b0d81b08b2ff78a6